docs/howto/partitions.rst

```diff
@@ -0,0 +1,193 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+.. Insert link below to new Snapcraft docs when this is merged and live:
+   https://github.com/canonical/snapcraft/issues/4857
+
+Applications implementing ``craft-application`` and its dependent suite of
+libraries can optionally make use of *partitions* when organizing the files in
+their output artifact.  Partitions are a generic concept, and may be used to
+implement a concrete feature in your application.  For instance, partitions may
+refer to:
+
+* Snapcraft's *components*
+* `Disk partitions <https://en.wikipedia.org/wiki/Disk_partitioning>`_ * Any
```

Comment:
A line break seems to be missing here.

Reply:
Good eye

Reply:
https://github.com/canonical/craft-application/pull/381